sdk/cosmosdb/cosmos/src/client/V2ChangeFeed/ChangeFeedEpkRange.ts

```diff
@@ -0,0 +1,327 @@
+    }
+
+    if (
+      canUseStartFromBeginning &&
+      changeFeedOptions.startFromBeginning !== undefined &&
+      !changeFeedOptions.startFromBeginning
+    ) {
+      throw new ErrorResponse(
+        "startFromBeginning must be true if no start time or continuation is specified."
+      );
+    }
+
+    this.queue = new FeedRangeQueue<ChangeFeedRange>();
+    this.continuationTokenMap = new ContinuationTokenMap(this.rId);
+
+    if (this.continuationToken) {
+      if (!this.continuationTokenRidMatchContainerRid()) {
+        throw new ErrorResponse("Container rid does not match with the continuation token rid");
+      } else {
+        for (const contToken of this.continuationToken.Continuation) {
+          const feedRange: ChangeFeedRange = new ChangeFeedRange(
+            contToken.minInclusive,
+            contToken.maxExclusive,
+            contToken.continuationToken
+          );
+
+          this.queue.enqueue(feedRange);
+          this.continuationTokenMap.addFeedRange(feedRange);
+        }
+      }
+    }
+  }
+
+  private continuationTokenRidMatchContainerRid(): boolean {
+    if (this.continuationToken.rid !== this.rId) {
+      return false;
+    }
+    return true;
+  }
+
+  public async fetchAllFeedRanges(): Promise<void> {
+    try {
+      const pkRanges = (
+        await this.partitionKeyRangeCache.onCollectionRoutingMap(this.url)
+      ).getOrderedParitionKeyRanges();
+      for (const pkRange of pkRanges) {
+        const feedRange: ChangeFeedRange = new ChangeFeedRange(
+          pkRange.minInclusive,
+          pkRange.maxExclusive,
+          ""
+        );
+        this.queue.enqueue(feedRange);
+        this.continuationTokenMap.addFeedRange(feedRange);
+      }
+    } catch (err) {
+      throw new Error(err.message);
+    }
+  }
+
+  public async fetchOverLappingFeedRanges(pkRange: PartitionKeyRange): Promise<void> {
+    const queryRange = new QueryRange(pkRange.minInclusive, pkRange.maxExclusive, true, false);
+    try {
+      const overLappingRanges = await this.partitionKeyRangeCache.getOverlappingRanges(
+        this.url,
+        queryRange
+      );
+
+      for (const overLappingRange of overLappingRanges) {
+        const feedRange: ChangeFeedRange = new ChangeFeedRange(
+          overLappingRange.minInclusive,
+          overLappingRange.maxExclusive,
+          ""
+        );
+        this.queue.enqueue(feedRange);
+        this.continuationTokenMap.addFeedRange(feedRange);
+      }
+    } catch (err) {
+      throw new ErrorResponse(err.message);
+    }
+  }
+  /**
+   * Gets a value indicating whether there are potentially additional results that can be retrieved.
+   *
+   * Initially returns true. This value is set based on whether there are any remaining unread epkRanges left in the queue.
+   *
+   * @returns Boolean value representing if whether there are potentially additional results that can be retrieved.
+   */
+  get hasMoreResults(): boolean {
```

Comment:
Change Feed always has more results (it is always true in .net). When will the queue be empty?